File: performance-tests/src/main/external/Fibonacci.js
function fibonacci(index) {
	if (index <= 1)
		return index;
	return fibonacci(index-1) + fibonacci(index-2);
}

for (var i=0; i<100; i++) {
	var startTime = process.hrtime.bigint();
	var result = fibonacci(12);
	var totalTime = process.hrtime.bigint() - startTime;
	console.log(totalTime);
}

    

File: performance-tests/src/main/external/Fibonacci.cpp
#include <iostream> 
#include <chrono> 

using namespace std;

int fibonacci(int n) {
    if (n <= 1)
        return n;
    return fibonacci(n - 1) + fibonacci(n - 2);
}

int main() {
    for (int i = 0; i < 100; i++) {
        auto startTime = chrono::high_resolution_clock::now().time_since_epoch();
        int result = fibonacci(12);
        auto totalTime = chrono::high_resolution_clock::now().time_since_epoch() - startTime;
        cout << totalTime << "\n";
    } 
}